### DISCUSSION TITLE
Issues using V1 Route Convention with Remix vite plugin

### DISCUSSION THREAD

**Discussion by aland-dev** [remix-run/remix#8372] [Q&A] [answered]
Hello there,

I'm trying to get [V1 route convention](https://github.com/remix-run/v1-compat-utils/tree/main/packages/v1-route-convention) to work with the new vite plugin but got the below error. Is this a remix issue or am I doing something wrong?

Here is the a the reproduction environment: https://stackblitz.com/edit/remix-vite-v1-routes

```bash
Error: No route matches URL "/example/new"
    at getInternalRouterError (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/router/router.ts:4483:5)
    at Object.query (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/router/router.ts:2933:19)
    at handleDocumentRequestRR (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/server-runtime/dist/server.js:180:35)
    at requestHandler (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/server-runtime/dist/server.js:105:24)
    at eval (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/express/dist/server.js:41:22)
No routes matched location "/example/new" 
ErrorResponseImpl {
  status: 404,
  statusText: 'Not Found',
  internal: true,
  data: 'Error: No route matches URL "/example/new"',
  error: Error: No route matches URL "/example/new"
      at getInternalRouterError (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/router/router.ts:4483:5)
      at Object.query (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/router/router.ts:2933:19)
      at handleDocumentRequestRR (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/server-runtime/dist/server.js:180:35)
      at requestHandler (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/server-runtime/dist/server.js:105:24)
      at eval (/home/projects/remix-vite-v1-routes/node_modules/@remix-run/express/dist/server.js:41:22)
}
```

**Comment by kiliman** [comment]
Change `ignoredRouteFiles` to ignore _all_ files. Remove the dot in the pattern. Otherwise Remix is processing the routes using the v2 convention.

**Reply by aland-dev** [reply]
Thank you for your answer, but it does not seem to work even with the dot remove in the pattern.

**Comment by hi-ogawa** [comment, marked as answer]
> I'm trying to get [V1 route convention](https://github.com/remix-run/v1-compat-utils/tree/main/packages/v1-route-convention) to work with the new vite plugin

It looks like you're not using Vite actually. Instead of `remix dev` and `remix build`, the scripts should look like this:

https://github.com/remix-run/remix/blob/36ccae4942d05cc21430d2d1ab6dc27f7b73fc71/templates/unstable-vite/package.json#L6-L7

Could you try starting from the official template? https://github.com/remix-run/remix/tree/main/templates/unstable-vite

**Reply by aland-dev** [reply]
Thanks, it worked :+1: it would be nice to have these in the migration guide

**Reply by Nedi11** [reply]
they are

**Reply by aland-dev** [reply]
Thanks indeed, https://remix.run/docs/en/main/future/vite#migrating-from-remix-app-server they've been updated recently :+1:

**Reply by Nedi11** [reply]
Has been there for 2 months https://github.com/remix-run/remix/blob/7ed944abda9368cbff6c4871802e50d0f89be699/docs/future/vite.md

**Reply by aland-dev** [reply]
need goggles

**Reply by ridoybevy** [reply]
tailwind css not working

**Reply by aland-dev** [reply]
https://remix.run/docs/en/main/future/vite#enable-tailwind-via-postcss

**Comment by hassankhanmv** [comment]
heres the complete answer, we are struggling too to create folder based routing,
first run this command
`npm i -D @remix-run/v1-route-convention`
and update your `vite.config.js`

add this line in remix({
```routes(defineRoutes) {
        // uses the v1 convention, works in v1.15+ and v2
        return createRoutesFromFolders(defineRoutes);
      },
```

my vite.config.js file lloks like this after updating it
```
import { vitePlugin as remix } from "@remix-run/dev";
import { defineConfig } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import {
  createRoutesFromFolders,
} from "@remix-run/v1-route-convention";

export default defineConfig({
  plugins: [
    remix({
      routes(defineRoutes) {
        // uses the v1 convention, works in v1.15+ and v2
        return createRoutesFromFolders(defineRoutes);
      },
      future: {
        v3_fetcherPersist: true,
        v3_relativeSplatPath: true,
        v3_throwAbortReason: true,
        v3_errorBoundary: true,
        v3_meta: true,
        v3_normalizeFormMethod: true,
        v3_routeConvention: false,
      },
    },
  ),
    tsconfigPaths(),
  ],
  
});
```

root directory
app/routes/inbound/inbound.jsx
now i can access my folders nested routes after this.

Best Regards
Eilmi | eilmi.com
